### PR TITLE
Drop doctests

### PR DESCRIPTION
**Pull request by MichaelXavier**
So I feel like I convered the meaningful text that was previously a doctest in the source code. As we discussed, I also removed all the code samples from the README which were never built and thus likely to lie. I didn't really port those into the examples file as a lot of them were not really sensible for a compiled example because they were just showing you how to use constructors. Still, I cut out some exposition as well in the process. I'm not sure what to do here. I could roll back changes to that file if anyone actually gets any use out of it but then we run the risk of the file not keeping up with the type changes.

### CODE REVIEW

**Comment by MichaelXavier** [comment]
I should also mention that this incorporates the semigroups PR I made and brings us to tests passing on nightlies so bloodhound should be able to get back into stackage.

File: stack-7.10.yaml [hunk 1]
```diff
@@ -1,13 +1,15 @@
-flags: {}
+flags:
+  bloodhound-examples:
+    werror: true
```
Comment (bitemyapp) [inline stack-7.10.yaml]: veddy good

**Comment by bitemyapp** [comment]
@MichaelXavier we need to drop the hspec upper bound as well, then it can get back into Stackage I think. I'll do that after merging this (when it passes)

**Comment by bitemyapp** [comment]
Restarted build, sigh.

File: stack-7.10.yaml (hunk 1 above)
Comment (MichaelXavier) [inline stack-7.10.yaml]: I'm not devious enough to put it on the whole project yet but give it time :P

**Comment by MichaelXavier** [comment]
@bitemyapp we're green!

**Pull request merged**